resources/assets/coffee/react/mp-history/main.coffee

```diff
@@ -42,18 +43,23 @@ class MPHistory.Main extends React.Component
 
       @setState initialLoaded: true
 
+  componentWillUnmount: ->
+    clearTimeout @timeout
+
   loadHistory: =>
+    return if _.last(@state.events)?.detail.type == 'match-disbanded'
+
     $.ajax laroute.route('matches.history', matches: @props.match.id, full: @props.full),
       method: 'GET'
       dataType: 'JSON'
       data:
-        since: @state.since
+        since: @minNextEventId()
 
     .done (data) =>
       return if _.isEmpty data.events.data
 
-      newEvents = _.concat @state.events, data.events.data
-      lastEvent = _.last(newEvents)
+      lastIndex = _.findLastIndex @state.events, (e) -> e.id < data.events.data[0].id
```

Comment:
this doesn't seem to be much faster than what I initially had since it still loops through the whole array ┐(･_･ )┌